ui/src/app/pages/analysis/leonardo-app-launcher.tsx

```diff
@@ -265,6 +291,10 @@ export const NotebookRedirect = fp.flow(
       return !!creating;
     }
 
+    private isCreatingTerminal() {
```

Comment:
Is "creating" accurate? What happens when you navigate away and come back - is history preserved?

Reply:
yes, sounds good